catalyst-gateway/bin/src/service/common/auth/rbac/token.rs

```diff
@@ -32,6 +38,10 @@ pub(crate) struct CatalystRBACTokenV1 {
     signature: Signature,
     /// Raw bytes of the token without the signature.
     raw: Vec<u8>,
+    /// A corresponded RBAC chain, constructed from the most recent data from the
+    /// database. Lazy initialized
+    /// TODO: make `RegistrationChain` clonable, remove Arc
```

Comment:
Why make it clonable and remove Arc? Cloning it would be expensive, and its immutable, so there is no need to remove the Arc it would be worse performance for no gain in functionality.  I personally think cloning a large structure by copying its memory is an antipattern when its easy to have a reference to a common immutable structure.  Its both cheap to clone, and memory efficient.  Neither of which is true for making a complete copy of a large structure in memory.

The only exception is if the Arc is embedded in the RegistrationChain directly, which is effectively the same thing as this, but cleaner to use.